model/project.rb

```diff
@@ -41,6 +41,24 @@ def path
     "/project/#{ubid}"
   end
 
+  def has_resources
+    access_tags_dataset.exclude(hyper_tag_table: [Account.table_name.to_s, Project.table_name.to_s, AccessTag.table_name.to_s]).count > 0
+  end
+
+  # Still thinking about the name, open for suggestions
+  def destroy_tags_and_hide
+    DB.transaction do
+      access_tags.each { |access_tag| access_tag.applied_tags_dataset.destroy }
+      access_tags_dataset.destroy
+      access_policies_dataset.destroy
+
+      # We still keep the project object for billing purposes.
```

Comment:
Are we still able to associate resources to projects while cutting invoices if we delete tags?

Reply:
The billing record model has "project_id" column